### PR TITLE
4770 Create a custom behavior for paraghrap which switch image align …

### PR DESCRIPTION
**Pull request by ChesterProg**
…+ install and configurate xdebug
https://redmine.devbranch.work/issues/4770
Create a custom behavior for paraghrap which switch image align
Install and configurate xdebug

### CODE REVIEW

File: web/modules/custom/paragraph_align/paragraph_align.routing.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+paragraph_align:
+  path: '/paragraph/page'
+  defaults:
+    _controller: '\Drupal\paragraph_align\Controller\ParagraphController::para'
+    _title: 'Welcome to My Module in Drupal 9'
```
Comment (Y0r) [inline web/modules/custom/paragraph_align/paragraph_align.routing.yml]: Is this title good for this page? In my opinion, it should describe related functionality.

File: web/modules/custom/paragraph_align/paragraph_align.routing.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+paragraph_align:
+  path: '/paragraph/page'
+  defaults:
+    _controller: '\Drupal\paragraph_align\Controller\ParagraphController::para'
```
Comment (Y0r) [inline web/modules/custom/paragraph_align/paragraph_align.routing.yml]: The controller name is okay, but the method name is weird.

File: web/modules/custom/paragraph_align/paragraph_align.routing.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+paragraph_align:
+  path: '/paragraph/page'
```
Comment (Y0r) [inline web/modules/custom/paragraph_align/paragraph_align.routing.yml]: Will be better: /admin/paragraphs/align

1) Management pages should be in /admin section
2) Path should be simple and describe the goal

File: web/modules/custom/paragraph_align/src/Controller/ParagraphController.php [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Drupal\paragraph_align\Controller;
```
Comment (Y0r) [inline web/modules/custom/paragraph_align/src/Controller/ParagraphController.php]: Why do you need this code? 
Better to remove useless code.

Comment (Y0r) [reply web/modules/custom/paragraph_align/src/Controller/ParagraphController.php]: Let's discuss it in monday

File: web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+/**
+ * @ParagraphsBehavior(
+ *   id = "dummy_image_and_text",
+ *   label = @Translation("Paragraph Image and Text settings"),
+ *   description = @Translation("Allows to select image size and position."),
+ *   weight = 0,
+ * )
+ */
+
+class imagealign extends ParagraphsBehaviorBase {
```
Comment (Y0r) [inline web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php]: Class name should be in CamelCase

File: web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\paragraphs\Annotation\ParagraphsBehavior;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\paragraphs\Entity\ParagraphsType;
+use Drupal\paragraphs\ParagraphInterface;
+use Drupal\paragraphs\ParagraphsBehaviorBase;
+
+/**
+ * @ParagraphsBehavior(
+ *   id = "dummy_image_and_text",
```
Comment (Y0r) [inline web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php]: Wrong identifier of the plugin. 
I think should be 'image_text_align' or something like that.

File: web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+ *   label = @Translation("Paragraph Image and Text settings"),
+ *   description = @Translation("Allows to select image size and position."),
```
Comment (Y0r) [inline web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php]: Nope, wrong description. Is it copy-paste code?

File: web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary(Paragraph $paragraph) {
+    $image_size = $paragraph->getBehaviorSetting($this->getPluginId(), 'image_size', '4_12');
+    $image_size_options = $this->getImageSizeOptions();
+    $image_position = $paragraph->getBehaviorSetting($this->getPluginId(), 'image_position', 'left');
+    $image_position_options = $this->getImagePositionOptions();
+
+    $summary = [];
```
Comment (Y0r) [inline web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php:83]: This line is useless.

File: web/themes/custom/bart/src/scss/MainCategoryPage.scss [hunk 9]
```diff
@@ -1,4 +1,4 @@
-// Header of category page.
+// Header of category page.7373
```
Comment (Y0r) [inline web/themes/custom/bart/src/scss/MainCategoryPage.scss]: Have a look at this comment I think something wrong with it.

**Review by Y0r — CHANGES_REQUESTED** [review]
A lot of problems, need to solve before merging.

**Comment by ChesterProg** [comment]
Resolved all issues.

File: web/modules/custom/paragraph_align/paragraph_align.routing.yml (hunk 3 above)
Comment (Y0r) [reply web/modules/custom/paragraph_align/paragraph_align.routing.yml]: Isn't solved

File: web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php (hunk 5 above)
Comment (Y0r) [reply web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php]: Read about CamelCase

File: web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php (hunk 6 above)
Comment (Y0r) [reply web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php]: isn't solved

File: web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php (hunk 7 above)
Comment (Y0r) [reply web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php]: isn't solved

File: web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php (hunk 8 above)
Comment (Y0r) [reply web/modules/custom/paragraph_align/src/Plugin/paragraphs/Behavior/imagealign.php:83]: Isn;t solved

File: web/modules/custom/paragraph_align/paragraph_align.routing.yml (hunk 1 above)
Comment (Y0r) [reply web/modules/custom/paragraph_align/paragraph_align.routing.yml]: Isn't solved

**Review by Y0r — APPROVED** [review]

**Pull request merged**